src/python/WMComponent/ErrorHandler/ErrorHandlerPoller.py

```diff
@@ -233,8 +238,13 @@ def readFWJRForErrors(self, jobList):
                     logging.debug(msg)
                     exhaustJobs.append(job)
                     continue
-
-                if len([x for x in report.getExitCodes() if x in self.exitCodes]):
+                
+                if self.reqAuxDB is None:
+                    exitCodesNoRetry = self.config.ErrorHandler.failureExitCodes
+                else:
+                    exitCodesNoRetry = self.reqAuxDB.getWMAgentConfig(self.config.Agent.hostName).get("NoRetryExitCodes", [])
+                
+                if len([x for x in report.getExitCodes() if x in exitCodesNoRetry]):
```

Comment:
Apparently the cache duration for ReqMgrAuxDB is set to 0.

Since `self.exitCodes` is initialized in the init method (with the agent config), how about we do the following:
* in the `algorithm` method, we check whether `self.reqAuxDB` is not None and in that case we update `self.exitCodes` with the `NoRetryExitCodes` content. Inside a try/except block such that it doesn't break the component and - if it fails, then it uses the previous list of exit codes.